internal/controller/linstorsatellite_test.go

```diff
@@ -186,48 +209,175 @@ var _ = Describe("LinstorSatelliteReconciler", func() {
 				g.Expect(ds.Spec.Template.Labels).To(HaveKeyWithValue("example.com/foo", "bar"))
 				g.Expect(ds.Spec.Template.Spec.HostNetwork).To(BeTrue())
 				g.Expect(ds.Spec.Template.Spec.Containers).NotTo(ContainElement(HaveField("Name", "drbd-reactor")))
-			}, DefaultTimeout, DefaultCheckInterval).Should(Succeed())
+			}).Should(Succeed())
 		})
 
-		Context("with additional finalizer", func() {
+		Context("with created Pod resource", func() {
+			var linstorClient *linstorhelper.Client
+
 			BeforeEach(func(ctx context.Context) {
-				err := k8sClient.Patch(ctx, &piraeusiov1.LinstorSatellite{
-					TypeMeta:   TypeMeta,
-					ObjectMeta: metav1.ObjectMeta{Name: ExampleNodeName, Finalizers: []string{"piraeus.io/test"}},
-				}, client.Apply, client.FieldOwner("test"), client.ForceOwnership)
+				var ds *appsv1.DaemonSet
+				Eventually(func() *appsv1.DaemonSet {
+					var current appsv1.DaemonSet
+					err := k8sClient.Get(ctx, types.NamespacedName{Namespace: Namespace, Name: "linstor-satellite." + ExampleNodeName}, &current)
+					if err != nil {
+						return nil
+					}
+					ds = &current
+					return ds
+				}).Should(Not(BeNil()))
+
+				pod := &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName:    "linstor-satellite-" + ExampleNodeName,
+						Namespace:       Namespace,
+						Labels:          ds.Spec.Template.Labels,
+						Annotations:     ds.Spec.Template.Annotations,
+						OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(ds, schema.FromAPIVersionAndKind("apps/v1", "DaemonSet"))},
+					},
+					Spec: ds.Spec.Template.Spec,
+				}
+
+				pod.Spec.NodeName = ExampleNodeName
+				err := k8sClient.Create(ctx, pod)
```

Comment:
Why does the test now have to create a satellite pod?

Reply:
We didn't need to create it up to now, because we did not have a controller API to begin with. Now that we have a "Controller", the operator needs the Pod to figure out the IP address of the satellite.